library/chacha20.c

```diff
@@ -516,6 +516,11 @@ static const size_t test_lengths[2] =
     375U
 };
 
+/* Make sure no other definition is already present. */
```

Comment:
A conforming C implementation isn't allowed to define `ASSERT`: it's in the application namespace.

In this C file, we only include a few standard library headers, some headers of our own, `config.h`, and possibly `platform_alt.h`. Where is `ASSERT` coming from on your build? None of them _should_ define `ASSERT`, but I realize that some embedded platforms define macros with “convenient” names that are officially reserved for applications.

Reply:
I get it from other libraries I have with the project.  I understand that these macros are only used for testing and maybe you don't want to #undef every macro.
It's just common to #undef common macros before defining them to avoid warnings.  I'm okay with it if we drop this patch.